feat: Allow org admins to set default workflow configs

SuaYoo opened the pull request:
Resolves https://github.com/webrecorder/browsertrix/issues/2017, follows https://github.com/webrecorder/browsertrix/pull/2031

### Changes

- Adds new org settings tab for updating crawl details
- Refactors `workflow-editor` to move out utility functions
- Updates user guide on org settings

### Manual testing

1. Log in as org admin
2. Go to Settings -> Crawl Workflows
3. Update settings and save
4. Go to New Workflow
5. Click through sections. Verify default values are set as expected
6. Save new workflow. Verify values save as expected
7. Go to Settings -> Crawl Workflows
8. Remove settings and save
9. Go to New Workflow. Verify default values are no longer set

SuaYoo committed:
add section

SuaYoo committed:
move file

SuaYoo committed:
refactor to btrix component

SuaYoo committed:
move to utils

SuaYoo committed:
move server defaults

SuaYoo committed:
add utils

SuaYoo committed:
delete unused file

SuaYoo committed:
update defaults

SuaYoo committed:
standardize default labels

SuaYoo committed:
add limits

SuaYoo committed:
refactor

SuaYoo committed:
add browser settings

SuaYoo committed:
add exclusions

SuaYoo committed:
update strings

SuaYoo committed:
add save

SuaYoo committed:
update category

Shrinks99 committed:
Adds billing tab to the org settings docs

Shrinks99 committed:
Adds Crawl Workflows tab to org settings

SuaYoo commented on docs/user-guide/org-settings.md:
```diff
@@ -2,9 +2,9 @@
+## General
 
-This tab lets you change the organization's name. This name must be unique.
+This tab lets you change the organization's name and URL identifier. Both of these must not be in use by any other orgs.
```
I like that this is reworded @Shrinks99 I think we could go a step further in editing this, since there's no way for users to know what the other orgs are currently, and so we don't give them a success path to pair with this warning. Something like:

`Change your organization's name and URL identifier in this tab. Choose an org name that's unique and memorable, like the name of your company or organization. Org name and URLs are unique to each Browsertrix instance (for example, on browsertrix.com) and you may be asked to change the org name if either are already in use by another org.`

Also, documentation best practices usually calls for an active voice, which we employ for the most part. Instead of "This tab lets you ..." I believe we should opt for a more humanist and user-centered "Change org name ...".

SuaYoo replied:
Updated here: https://github.com/webrecorder/browsertrix/pull/2020/commits/27361672dcb79c68d9e0427fb18e877e12b5a165

SuaYoo commented on docs/user-guide/org-settings.md:
```diff
@@ -31,6 +31,10 @@ Sent invites can be invalidated by pressing the trash button in the relevant _Pe
 
+## Crawl Workflows
+
+This tab lets admins change the default suggested values for new crawl workflows. These defaults are not enforced and users can set their own values, or revert them back to the server-set defaults and limits.
```
Similar feedback here, for more user-centered docs in active voice, with less punitive language (e.g. "not enforced"):

`Set and update default suggested settings for new crawl workflows. When creating a new workflow, org members will see that the form is pre-filled with default values. Org members can still change or remove these settings when configuring the workflow. Removing a default setting will revert the setting back to Browsertrix defaults.`

Also, I don't think we need to mention that they're not enforced since we say upfront that they're suggested.

SuaYoo committed:
convert exclusion table to tailwind component

SuaYoo committed:
make exclusion table uncontrolled

SuaYoo committed:
save crawl defaults

SuaYoo committed:
enable unsetting language

SuaYoo committed:
clear last exclusion instead of deleting

SuaYoo committed:
use org defaults

SuaYoo committed:
update org settings user guide

SuaYoo committed:
update default fields

SuaYoo committed:
standardize info col

SuaYoo committed:
fix resetting browser profile

SuaYoo committed:
fix types

SuaYoo committed:
fix initial values

SuaYoo committed:
update browser profile label

SuaYoo committed:
update placeholder and help text

SuaYoo committed:
move app settings

SuaYoo committed:
pass settings to app

SuaYoo commented:
@Shrinks99 FYI moved docs changes over to https://github.com/webrecorder/browsertrix/pull/2050

SuaYoo committed:
rename layout folder

SuaYoo committed:
uncomment docs

tw4l commented on frontend/src/pages/org/settings/components/crawl-workflows.ts:
```diff
@@ -0,0 +1,336 @@
+            <sl-button type="submit" size="small" variant="primary">
+              ${msg("Save Changes")}
+            </sl-button>
+          </footer>
```
Hardly a blocker, but it could be nice to also have a "Reset" button here somewhere, perhaps with dialog confirmation before applying. Right now if you want to clear all the values and start over it's a lot of clicks/manual deletion.

ikreymer replied:
Would that reset to the previously saved defaults, or remove all defaults altogether? The former can just be accomplished by refreshing the page..
It seems like users might be changing these incrementally, so maybe not too big of a deal either way.

tw4l replied:
I was imagining a function just to clear the form altogether, but it's true that in practice that probably wouldn't be done much so unless we get user requests, just ignore this!

tw4l commented on frontend/src/strings/crawl-workflows/infoText.ts:
```diff
@@ -0,0 +1,63 @@
+
+type Field = keyof FormState;
+
+const infoText: Partial<Record<Field, string | TemplateResult>> = {
```
Nice refactoring!

tw4l reviewed (approved):
Really nice work! Feature's working great in the frontend the reorganization of some of the workflow editor code and types is great.

ikreymer committed:
use 'Crawling Defaults' as the name of tab / component
rename settings crawl-workflows -> crawling defaults
rename component to btrix-org-settings-crawling-defaults

ikreymer commented:
Renamed tab / component to 'Crawling Defaults' for consistency, to make it clear this is about defaults for crawl workflows and not to be confused with crawl workflows themselves.

Shrinks99 commented on docs/user-guide/org-settings.md:
```diff
@@ -12,9 +12,9 @@ Change your organization's name and URL identifier in this tab. Choose an org na
 View and manage your org's current payment plan and associated quotas. Usage history statistics for previous months are shown here to better inform your plan and quota requirements.
 
-<!-- ## Crawl Workflows
+## Crawl Workflows
```
```suggestion
## Crawling Defaults
```

Shrinks99 commented on docs/user-guide/org-settings.md:
```diff
@@ -12,9 +12,9 @@ Change your organization's name and URL identifier in this tab. Choose an org na
+## Crawl Workflows
 
-Set default suggested settings for all new crawl workflows in this tab. When creating a new workflow, org members will see the form pre-filled with default values. Org members can still change or remove these settings when configuring the workflow. Removing a default setting will revert the setting back to Browsertrix defaults. -->
+Set default suggested settings for all new crawl workflows in this tab. When creating a new workflow, org members will see the form pre-filled with default values. Org members can still change or remove these settings when configuring the workflow. Removing a default setting will revert the setting back to Browsertrix defaults.
```
```suggestion
Set default suggested settings for all new crawl workflows. When creating a new workflow, org members will see the form pre-filled with these default values. Org members can still change or remove these settings when configuring the workflow. Removing a default setting will revert the setting back to Browsertrix' defaults.
```

Shrinks99 reviewed (approved):
Minor docs fixes, other than this looks great!

SuaYoo committed:
Apply suggestions from code review

Co-authored-by: Henry Wilkinson <henry@wilkinson.graphics>
